packages/backend/migrations/20240820161040_add_ledger_transfer_constraint.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+  return knex.schema.table('ledgerTransfers', function (table) {
+    table.check(
+      `("state" != 'PENDING') OR ("expiresAt" IS NOT NULL)`,
```

Comment:
Does the new query error if some record doesnt satisfy the constraint? Just wondering if we can guaruntee all existing records will pass this constraint... migration will fail otherwise. Im also wondering if the error is intelligible if we try to add one that doesnt pass this constraint. Not really sure where that would happen or how it should surface...

Reply:
Checking rafiki money would be a good test, but I think it's not actually something that should have ever happened, this is how we map transfers to create them:

https://github.com/interledger/rafiki/blob/750f4148618990b3b0d85d1d987b39fc46c1ac19/packages/backend/src/accounting/psql/ledger-transfer/index.ts#L326-L342

i.e. if the timeout is provided, we set PENDING with expiryAt always

Reply:
rafiki.money db:

```sh
rafiki_backend=# select count(*) from "ledgerTransfers" where state = 'PENDING' and "expiresAt" is null;
 count 
-------
     0
(1 row)

rafiki_backend=# 
```